### PR TITLE
Fix SYSLIB obsoletion warnings in tests

### LINKED ISSUE #39976: Obsolete warnings on clean build
*Initial cost estimate*: 2 days
*Initial contacts*: @trylek, @ViktorHofer 

#39269 introduced warnings to the full build.

For example https://dev.azure.com/dnceng/public/_build/results?buildId=746713&view=results

Shows 18 warnings which seem to be test code:
```
LowLatencyTest.cs(111,13): warning SYSLIB0004: 'RuntimeHelpers.PrepareConstrainedRegions()' is obsolete: 'The Constrained Execution Region (CER) feature is not supported.'
CoreCLR Common Pri0 Test Build AnyOS AnyCPU checked • Build managed test components
TestInALC.cs(13,77): warning SYSLIB0012: 'Assembly.CodeBase' is obsolete: 'CodeBase and EscapedCodeBase are only included for .NET Framework compatibility. Use Location instead.'
CoreCLR Common Pri0 Test Build AnyOS AnyCPU checked • Build managed test components
RunInALC.cs(42,73): warning SYSLIB0012: 'Assembly.CodeBase' is obsolete: 'CodeBase and EscapedCodeBase are only included for .NET Framework compatibility. Use Location instead.'
CoreCLR Common Pri0 Test Build AnyOS AnyCPU checked • Build managed test components
TestInALC.cs(13,77): warning SYSLIB0012: 'Assembly.CodeBase' is obsolete: 'CodeBase and EscapedCodeBase are only included for .NET Framework compatibility. Use Location instead.'
CoreCLR Common Pri0 Test Build AnyOS AnyCPU checked • Build managed test components
TestInALC.cs(13,77): warning SYSLIB0012: 'Assembly.CodeBase' is obsolete: 'CodeBase and EscapedCodeBase are only included for .NET Framework compatibility. Use Location instead.'
CoreCLR Common Pri0 Test Build AnyOS AnyCPU checked • Build managed test components
GitHub_25468.cs(82,9): warning SYSLIB0011: 'BinaryFormatter.Serialize(Stream, object)' is obsolete: 'BinaryFormatter serialization is obsolete and should not be used. See https://aka.ms/binaryformatter for more information.'
CoreCLR Common Pri0 Test Build AnyOS AnyCPU checked • Build managed test components
GitHub_25468.cs(85,24): warning SYSLIB0011: 'BinaryFormatter.Deserialize(Stream)' is obsolete: 'BinaryFormatter serialization is obsolete and should not be used. See https://aka.ms/binaryformatter for more information.'
CoreCLR Common Pri0 Test Build AnyOS AnyCPU checked • Build managed test components
test22888.cs(68,73): warning SYSLIB0012: 'Assembly.CodeBase' is obsolete: 'CodeBase and EscapedCodeBase are only included for .NET Framework compatibility. Use Location instead.'
CoreCLR Common Pri0 Test Build AnyOS AnyCPU checked • Build managed test components 
IpcUtils.cs(73,55): warning SYSLIB0012: 'Assembly.CodeBase' is obsolete: 'CodeBase and EscapedCodeBase are only included for .NET Framework compatibility. Use Location instead.'
CoreCLR Common Pri0 Test Build AnyOS AnyCPU checked • Build managed test components
LowLatencyTest.cs(111,13): warning SYSLIB0004: 'RuntimeHelpers.PrepareConstrainedRegions()' is obsolete: 'The Constrained Execution Region (CER) feature is not supported.' Mono Common Pri0 Test Build AnyOS AnyCPU release • Build managed test components
TestInALC.cs(13,77): warning SYSLIB0012: 'Assembly.CodeBase' is obsolete: 'CodeBase and EscapedCodeBase are only included for .NET Framework compatibility. Use Location instead.' Mono Common Pri0 Test Build AnyOS AnyCPU release • Build managed test components
RunInALC.cs(42,73): warning SYSLIB0012: 'Assembly.CodeBase' is obsolete: 'CodeBase and EscapedCodeBase are only included for .NET Framework compatibility. Use Location instead.' Mono Common Pri0 Test Build AnyOS AnyCPU release • Build managed test components
TestInALC.cs(13,77): warning SYSLIB0012: 'Assembly.CodeBase' is obsolete: 'CodeBase and EscapedCodeBase are only included for .NET Framework compatibility. Use Location instead.' Mono Common Pri0 Test Build AnyOS AnyCPU release • Build managed test components
TestInALC.cs(13,77): warning SYSLIB0012: 'Assembly.CodeBase' is obsolete: 'CodeBase and EscapedCodeBase are only included for .NET Framework compatibility. Use Location instead.' Mono Common Pri0 Test Build AnyOS AnyCPU release • Build managed test components
GitHub_25468.cs(82,9): warning SYSLIB0011: 'BinaryFormatter.Serialize(Stream, object)' is obsolete: 'BinaryFormatter serialization is obsolete and should not be used. See https://aka.ms/binaryformatter for more information.' Mono Common Pri0 Test Build AnyOS AnyCPU release • Build managed test components
GitHub_25468.cs(85,24): warning SYSLIB0011: 'BinaryFormatter.Deserialize(Stream)' is obsolete: 'BinaryFormatter serialization is obsolete and should not be used. See https://aka.ms/binaryformatter for more information.' Mono Common Pri0 Test Build AnyOS AnyCPU release • Build managed test components
test22888.cs(68,73): warning SYSLIB0012: 'Assembly.CodeBase' is obsolete: 'CodeBase and EscapedCodeBase are only included for .NET Framework compatibility. Use Location instead.' Mono Common Pri0 Test Build AnyOS AnyCPU release • Build managed test components
IpcUtils.cs(73,55): warning SYSLIB0012: 'Assembly.CodeBase' is obsolete: 'CodeBase and EscapedCodeBase are only included for .NET Framework compatibility. Use Location instead.' Mono Common Pri0 Test Build AnyOS AnyCPU release • Build managed test components
```

cc @jeffhandley

### PR DESCRIPTION
**Pull request by jeffhandley**
Fixes #39976

PR #39269 introduced warnings to the full build during .NET 5.0 when the `SYSLIB` obsoletions were introduced. PR #42272 configured warnings to be treated as errors and achieved a clean build, but the `SYSLIB` diagnostics were simply suppressed. This PR addresses the `SYSLIB` diagnostics through targeted changes or in-place suppressions as appropriate and removes the `<NoWarn>` for them.

### CODE REVIEW

File: src/tests/Interop/COM/NETClients/Primitives/TestInALC.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@ class Test
     {
         static int Main(string[] args)
         {
-            string currentAssemblyDirectory = Path.GetDirectoryName(new Uri(Assembly.GetExecutingAssembly().CodeBase).AbsolutePath);
+            string currentAssemblyDirectory = Path.GetDirectoryName(new Uri(Assembly.GetExecutingAssembly().Location).AbsolutePath);
```
Comment (jkotas) [inline src/tests/Interop/COM/NETClients/Primitives/TestInALC.cs:13]:
```suggestion
            string currentAssemblyDirectory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
```
`AssemblyCodeBase` was URL. `Assembly.Location` is file path.

Comment (jeffhandley) [reply src/tests/Interop/COM/NETClients/Primitives/TestInALC.cs:13]: Thanks. I verified that wrapping each in `new Uri` got the same net result, but then I failed to catch that the `Uri` was an inconvenient detail here.

Comment (jkotas) [reply src/tests/Interop/COM/NETClients/Primitives/TestInALC.cs:13]: Multiple files in this PR have the same issue.

Comment (jeffhandley) [reply src/tests/Interop/COM/NETClients/Primitives/TestInALC.cs:13]: Thanks; I'll fix 'em all

**Review by safern — APPROVED** [review]
LGTM.

**Pull request merged**